export.md

```diff
@@ -57,6 +57,34 @@ Export data from a FHIR server whether or not it is associated with a patient. T
 
   Note: Some implementations may limit the resources returned to specific subsets of FHIR like those defined in the [Argonaut Implementation Guide](http://www.fhir.org/guides/argonaut/r2/)
 
+##### Experimental Query Parameters
+
+As a community, we've identified use cases for finer-grained, client-specified filtering. For example, some clients may want to retrieve only active prescriptions (rather than historical prescriptions), or only laboratory observations (rather than all observations). We have considered several approaches to finer-grained filtering, including FHIR's `GraphDefinition`, the Clinical Query Language, and FHIR's REST API search parameters. We expect this will be an area of active exploration, so for the time being we're defining an experimental syntax based on search parameters that works side-by-side with our coarse-grained `_type`-based filtering.
+
+To request finer-grained filtering, a client can supply a `_typeFilter` parameter alongside the `_type` parameter. The value of the `_typeFilter` parameter is a comma-separated list of FHIR REST API queries that further restrict the results of the query. Understanding `_typeFilter` is optional for servers; clients should be robust to servers that ignore `_typeFilter`. To avoid potentially surprising results, `_typeFilter` must not be used in combination with `_since`. (The example below shows how time-based constraints can still be accomplished by pushing them into the `_typeFilter` if needed.)
```

Comment:
@isaacvetter @jmandel , it seems to me that the correct behavior around conflicts between the `_since` and `_type` parameters, and the `_typeFilter` parameter is for a server to always follow `_since` and `_type` restrictions, and then perform additional filtering of the result set if it supports the `_typeFilter` parameter (essentially returning the the intersection of the searches). That way, the same query sent to servers with and without `_typeFilter` support would never result in the server with support for the parameter returning more data than the server without support, and a client that does the `_typeFilter` filtering on the client side as it writes results will always return the same results as when the filtering is performed on the server.

Reply:
Hey @gotdan ,

1) If a MedicationRequest was last updated on 20180702, how should a FHIR server respond to a bulk data request like this:
```
Group$export?_since=20180703&_type=MedicationRequest&_typeFilter=MedicationRequest?_lastUpdated=gt2018-07-01
```

The "intersection" model implies multiple passes over the data, an implementation would be better served by interpreting _since into resource-level parameters (most likely _lastUpdated), in which case, we'd need guidance on handling conflicts between _since and _lastUpdated. 

2) The other potential source of problems is if a FHIR server didn't universally support ```_since``` across all resources. Would it be a problem if, for example, the server returned all active AllergeryIntolerances even with ```_lastUpdated``` values before the ```_since``` datetime?

Reply:
Hi @isaacvetter ,

I think we agree that we need to provide guidance on the use case you describe in #1 - I'm just suggesting that we have `_since` take precedence to ensure that clients which apply the `_typeFilter` filters on the client side against a server that doesn't support the parameter will end up  with the same data as that coming from  a server that does support the parameter. Also, I think your implementation note is right - I mentioned the two stage filtering as a way of describing the behavior rather than as any sort of recommendation :).

#2 brings up a good question. My understanding is that servers need to support `_since` (when it's provided) on all resource types, but I realize this is not explicit in the docs. Is there a strong use case to make support across all resources optional and if so, would we need some sort of conformance statement by servers to describe their capabilities (not ideal)?

Reply:
Re `_since` at the top level vs  `_lastUpdate` at an individual resource level, there really shouldn't be an opportunity for "conflict", because `_since={x}` translates to `_lastUpdated=gt{x]`. If you have both:

* `_lastUpdated=gt{x}` (from the top-level query *and*
* `_lastUpdated=gt{y}` (from a resource-type-specific typeFilter query)

... then this behaves exactly like `_lastUpdated=gt{max(x,y)}`. There should never be ambiguity or conflict.

Re: support for `_since`, my read is that servers must support this parameter in the sense that they must not reject a query for including a `_since`. Servers might always decide to return more results than the client asked for (i.e. ignoring `_since` is semantically defensible, but rejecting the query is not).

Reply:
I like the simplicity of a `_lastUpdated=gt{max(x,y)}` algorithm, but I worry it's going to cause confusion in the case of `y < x`, where servers that support `_typeFilter` will return more data than servers that don't. Having `_since` always take precedence would head off this edge case, and make `_typeFilter` a modifier of the other parameters rather than an alternate approach.

Re: support for `_since` I agree that's the best approach - should we make it explicit in the documentation?

Reply:
Creating special-case rules here seems odd to me; but maybe @gotdan all you're saying is that the server should perform the normal `_since`-based filtering in addition to any `_typeFilter`-based filtering if required? If so, I think that's the same thing I'm saying.

Given all of the confusion, what I'm going to do for now is change the prohibition about `_since` + `_typeFilter` to a warning for clients, and see if we can wrap it up there.